app/controllers/katello/api/v2/content_view_components_controller.rb

```diff
@@ -0,0 +1,85 @@
+module Katello
+  class Api::V2::ContentViewComponentsController < Api::V2::ApiController
+    before_action :find_composite_content_view
+    before_action :find_content_view_component, :only => [:show, :update, :destroy]
+
+    wrap_parameters :include => %w(composite_content_view_id content_view_version_id content_view_id latest)
+
+    api :GET, "/content_views/:composite_content_view_id/content_view_components",
+        N_("List components attached to this content view")
+    param :composite_content_view_id, :identifier, :desc => N_("composite content view identifier"), :required => true
+    def index
+      results = ContentViewComponent.where(:composite_content_view_id => @view)
+      collection = {
+        :results  => results.uniq,
+        :subtotal => results.count,
+        :total    => results.count
+      }
+      respond :collection => collection
+    end
+
+    api :POST, "/content_views/:composite_content_view_id/content_view_components",
+        N_("Add a component to the content view")
+    param :composite_content_view_id, :identifier, :desc => N_("composite content view identifier"), :required => true
+    param :content_view_version_id, :identifier, :desc => N_("identifier of the version of the component content view")
+    param :content_view_id, :identifier,
+          :desc => N_("content view identifier of the component who's latest version is desired")
+    param :latest, :bool, :desc => N_("true if the latest version of the components content view is desired")
+    def create
+      options = {:composite_content_view => @view}
+      options[:content_view_version_id] = component_params[:content_view_version_id] if component_params.key?(:content_view_version_id)
+      options[:content_view_id] = component_params[:content_view_id] if component_params.key?(:content_view_id)
+      options[:latest] = ::Foreman::Cast.to_bool(component_params[:latest]) if component_params.key?(:latest)
+      respond resource: ContentViewComponent.create!(options)
+    end
+
+    api :GET, "/content_views/:composite_content_view_id/content_view_components/:id", N_("Show a content view component")
+    param :composite_content_view_id, :number, :desc => N_("composite content view numeric identifier"), :required => true
```

Comment:
I guess my point is that it would be nice to have a `/content_view_components/:id` route so I don't have to also pass the content view id every time.